src/thingset.h

```diff
@@ -401,24 +401,24 @@ struct ts_data_object {
     /**
      * One of TS_TYPE_INT32, _FLOAT, ...
      */
-    const uint8_t type;
+    const uint32_t type : 4;
 
     /**
      * Exponent (10^exponent = factor to convert to SI unit) for decimal fraction type,
```

Comment:
Could you improve the documentation a little bit in the course of the change. It took me quite a time to understand that this are three different usages of the detail information depending on the type of data object. And I'm stiill not shure how to handle the exponent of the decimal fraction type. It seems you expect this to be a fixed value and not something that can be adapted acc. to the value.

For doxygen it would be nice to have a brief description and a detailed description.

Reply:
I have updated the documentation, please have a look.